docs/hooks/overview.md

```diff
@@ -67,7 +67,9 @@ This allows configuration-as-code inside lakeFS, where Action files are declarat
 {: .no_toc }
 
 An **Action** is a list of Hooks with the same trigger configuration, i.e. an event will trigger all Hooks under an Action or none at all.
-The Hooks under an Action are ordered and so is their execution. A Hook will only be executed if all the previous Hooks that were triggered with it had passed.
+The Hooks under an Action are ordered and so is their execution.
+Before each hook execution the `if` boolean expression is evaluated. The expression can use `success()` which evaluate to true in case of no error was occured while running the hoook's actions. It can also use `failure()` which evaluate to true when previous step failed - can be used to report failures.
```

Comment:
```suggestion
Before each hook execution the `if` boolean expression is evaluated. The expression can use `success()` which evaluate to true in case no error has occurred while running the hook's actions. It can also use `failure()` which evaluates to true if a previous step failed and can be used to report failures as well.
```